### PR TITLE
`!!reminders` feature

### PR DESCRIPTION
**Pull request by Infernum1**

### CODE REVIEW

File: bot/cogs/commands/useful.py [hunk 1]
```diff
@@ -663,13 +664,29 @@ async def remind_me(self, ctx: Ctx, duration_str: str, *, reminder: str):
...
             )
         )
 
+    @commands.command(name="reminders")
+    @commands.cooldown(1, 2, commands.BucketType.user)
+    async def reminders_list(self, ctx: Ctx):
+        user_reminders = await self.db.fetch_user_reminders(ctx.author.id)
+        embed = discord.Embed(color=self.bot.embed_color)
+        embed.set_author(name=f"{ctx.author.display_name}'s reminders", icon_url=ctx.author.avatar.url)
+        for i in user_reminders:
```
Comment (Iapetus-11) [inline bot/cogs/commands/useful.py]: Can we make this loop variable more descriptive? How about `for reminder in user_reminders`?

Comment (Iapetus-11) [inline bot/cogs/commands/useful.py]: Can we add a newline inbetween line 680 and 681? Spacing out code makes it more readable.

File: bot/cogs/commands/useful.py [hunk 2]
```diff
@@ -663,13 +664,29 @@ async def remind_me(self, ctx: Ctx, duration_str: str, *, reminder: str):
...
+            unix_timestamp = format_dt(i['at'], style = "R")
+            reminder = f"{i['reminder']}"
+            if len(reminder) > 75:
+                reminder = reminder[:75] + '...'
+            id = i['id']
+            embed.add_field(name=f'`#{id}` - {unix_timestamp}', value=reminder, inline=False)
```
Comment (Iapetus-11) [inline bot/cogs/commands/useful.py]: There's two issues with this
1. `id` is a built in function and you're overriding it by setting it to `i['id']`
2. You could just do something like:
```suggestion
            embed.add_field(name=f'`#{i["id"]}` - {unix_timestamp}', value=reminder, inline=False)
```

File: bot/cogs/commands/useful.py [hunk 3]
```diff
@@ -663,13 +664,29 @@ async def remind_me(self, ctx: Ctx, duration_str: str, *, reminder: str):
...
+    @commands.cooldown(1, 2, commands.BucketType.user)
+    async def reminders_list(self, ctx: Ctx):
+        user_reminders = await self.db.fetch_user_reminders(ctx.author.id)
+        embed = discord.Embed(color=self.bot.embed_color)
+        embed.set_author(name=f"{ctx.author.display_name}'s reminders", icon_url=ctx.author.avatar.url)
+        for i in user_reminders:
+            unix_timestamp = format_dt(i['at'], style = "R")
+            reminder = f"{i['reminder']}"
+            if len(reminder) > 75:
+                reminder = reminder[:75] + '...'
```
Comment (Iapetus-11) [inline bot/cogs/commands/useful.py]: [There's a function for shortening text you can use](https://github.com/Iapetus-11/Villager-Bot/blob/main/bot/utils/misc.py#L348)

File: bot/cogs/commands/useful.py [hunk 4]
```diff
@@ -663,13 +664,29 @@ async def remind_me(self, ctx: Ctx, duration_str: str, *, reminder: str):
...
 
+    @commands.command(name="reminders")
+    @commands.cooldown(1, 2, commands.BucketType.user)
+    async def reminders_list(self, ctx: Ctx):
+        user_reminders = await self.db.fetch_user_reminders(ctx.author.id)
+        embed = discord.Embed(color=self.bot.embed_color)
+        embed.set_author(name=f"{ctx.author.display_name}'s reminders", icon_url=ctx.author.avatar.url)
+        for i in user_reminders:
+            unix_timestamp = format_dt(i['at'], style = "R")
+            reminder = f"{i['reminder']}"
```
Comment (Iapetus-11) [inline bot/cogs/commands/useful.py]: `i['reminder']` is already a string? Why is this necessary?

File: bot/cogs/commands/useful.py [hunk 5]
```diff
@@ -663,13 +664,29 @@ async def remind_me(self, ctx: Ctx, duration_str: str, *, reminder: str):
...
         )
 
+    @commands.command(name="reminders")
+    @commands.cooldown(1, 2, commands.BucketType.user)
+    async def reminders_list(self, ctx: Ctx):
+        user_reminders = await self.db.fetch_user_reminders(ctx.author.id)
+        embed = discord.Embed(color=self.bot.embed_color)
+        embed.set_author(name=f"{ctx.author.display_name}'s reminders", icon_url=ctx.author.avatar.url)
+        for i in user_reminders:
+            unix_timestamp = format_dt(i['at'], style = "R")
```
Comment (Iapetus-11) [inline bot/cogs/commands/useful.py]: Generally when using kwargs (keyword arguments, like `f(a='abc')`) you don't have spaces on either side of the equal sign.

**Review by Iapetus-11 — CHANGES_REQUESTED** [review]

File: bot/cogs/core/database.py [hunk 6]
```diff
@@ -42,6 +42,9 @@ async def populate_caches(self):
     async def fetch_user_reminder_count(self, user_id: int) -> int:
         return await self.db.fetchval("SELECT COUNT(*) FROM reminders WHERE user_id = $1", user_id)
 
+    async def fetch_user_reminders(self, user_id: int):
```
Comment (Iapetus-11) [inline bot/cogs/core/database.py]: Please typehint the return of this function

**Review by Iapetus-11 — CHANGES_REQUESTED** [review]

File: bot/cogs/core/database.py (hunk 6 above)
Comment (Infernum1) [reply bot/cogs/core/database.py]: I should type hint it to a list right?

File: bot/cogs/commands/useful.py (hunk 4 above)
Comment (Infernum1) [reply bot/cogs/commands/useful.py]: oh yeah I originally added this to add an ellipses but forgot to remove it, will fix

File: bot/cogs/core/database.py [hunk 7]
```diff
@@ -42,6 +42,9 @@ async def populate_caches(self):
     async def fetch_user_reminder_count(self, user_id: int) -> int:
         return await self.db.fetchval("SELECT COUNT(*) FROM reminders WHERE user_id = $1", user_id)
 
+    async def fetch_user_reminders(self, user_id: int) -> list:
```
Comment (Iapetus-11) [inline bot/cogs/core/database.py]: Yeah it's returning a list, but a list of what?

https://mypy.readthedocs.io/en/stable/generics.html#defining-generic-classes

**Review by Iapetus-11 — CHANGES_REQUESTED** [review]
I did this review on mobile so I'm going to look over it again, but for now except that one comment it looks good!

File: bot/data/text/en.json [hunk 8]
```diff
@@ -1726,7 +1726,8 @@
           "xe": "Minor bug fixes and feature additions",
           "radar": "Minor bug fixes and feature additions",
           "kuro": "Helps with multiple translations of Villager Bot",
-          "hilt": "Helps with multiple translations of Villager Bot"
+          "hilt": "Helps with multiple translations of Villager Bot",
+          "infernum": "Miscellaneous bug fixes and feature additions and reminders system improvements"
```
Comment (Iapetus-11) [inline bot/data/text/en.json:1730]: You'll need to get the translation for this

**Review by Iapetus-11 — CHANGES_REQUESTED** [review]
You'll also need the FR translation for the reminder changes

**Review by Iapetus-11 — APPROVED** [review]
